compiler/src/dotty/tools/dotc/core/DenotTransformers.scala

```diff
@@ -60,11 +60,14 @@ object DenotTransformers {
   /** A transformer that only transforms SymDenotations */
   trait SymTransformer extends DenotTransformer {
 
+  	/** Tramsform the info of a denotation that is not a Symdenotation */
+    def transformInfo(tp: Type, sym: Symbol)(implicit ctx: Context): Type = tp
```

Comment:
I would call it `transformNonSymInfo` to avoid confusion. Or perhaps just make ElimOpaque a regular DenotTransformer, we can always add another kind of Transformer later if we find more uses for it.

Reply:
OK for `transformNonSymInfo`. I think it's better to leave it there to make it clear that something has to be done if a SymTranformer also changes the infos of symbols.